contracts/bid/src/hosted/transaction.rs

```diff
@@ -0,0 +1,286 @@
+    ///
+    /// This function will first of all, verify that the Bid is correct by
+    /// verifying the BidCorrectness Proof.
+    /// Then it will include the Bid into the PoseidonTree of the contract and
+    /// link the One-Time identity of the user to the index that the bid
+    /// occupies in the tree. Finally it will execute an inter-contract call
+    /// sending the `spending_proof` and a `note` to the transfer contract.
+    /// Which will execute the transaction of Dusk to the contract account.
+    pub fn bid(
+        &mut self,
+        mut bid: Bid,
+        correctness_proof: Vec<u8>,
+        _spending_proof: Vec<u8>,
+    ) -> (bool, usize) {
+        // Setup error flag to false
+        let mut err_flag = false;
+
+        // Verify proof of Correctness of the Bid.
+        if !fake_abi::verify_proof(
+            correctness_proof,
+            crate::BID_CORRECTNESS_VK.to_vec(),
+            b"bid-correctness".to_vec(),
+            PublicInput::AffinePoint(bid.commitment(), 0, 0)
+                .to_bytes()
+                .to_vec(),
+        ) {
+            err_flag = true;
+            return (err_flag, usize::MAX);
+        }
+
+        // Obtain the current block_height.
+        let block_height = dusk_abi::block_height();
+        // Compute maturity & expiration periods
+        let expiration = block_height + MATURITY_PERIOD + EXPIRATION_PERIOD;
+        let eligibility = block_height + MATURITY_PERIOD;
+
+        // Mutate the Bid and add the correct timestamps.
+        bid.set_eligibility(eligibility);
+        // FIXME: This should not be needed. We should have a better API in blindbid.
+        //  Since the API for blindbid was decided to be set as `extend_expiration` instead of
+        // `set_expiration`. We now are forced to ensure that the expiration is 0 here and then, we
+        // sum `expiration` to it.
+        assert!(bid.expiration() == 0u64);
+        bid.extend_expiration(expiration);
+
+        // Panic and stop the execution if the same one-time-key tries to
+        // bid more than one time.
+        let idx = if self
+            .key_idx_map()
+            // If no entries are found for this PK, add it to the map and the
+            // tree
+            .get(*bid.stealth_address().pk_r())
+            .unwrap()
+            .is_none()
+        {
+            // Append Bid to the tree and obtain the position of it.
+            let idx = self.tree_mut().push(BidLeaf(bid));
+            // Link the One-time PK to the idx in the Map
+            // Since we checked on the `get` call that the value was not
+            // inside, there's no need to check that this
+            // returns `Ok(None)`. So we just unwrap
+            // the `Result` and keep the `Option` untouched.
+            self.key_idx_map_mut()
+                .insert(*bid.stealth_address().pk_r(), idx)
+                .unwrap();
+            idx
+        } else {
+            err_flag = true;
```

Comment:
I believe it should be `true` in case of success, and not the opposite

Reply:
kk if that's the convention I'll change it